meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -421,6 +430,23 @@ public void handleConnectStateChanged(MetaClientInterface.ConnectState prevState
             LOG.info("Participant {} already in leader group {}.", _participant, leaderPath);
           }
         }
+        // resubscribe the  re-elect listener
+        for (String leaderPath : _leaderGroups) {
+            LOG.info("Resubscribe re-elect listener for leaderPath {}.", leaderPath);
+          _metaClient.subscribeDataChange(leaderPath + LEADER_ENTRY_KEY, _reElectListener, false);
+        }
+
+        // resubscribe to leader entry change since we are reconnected
+        for (Map.Entry<String, Set<LeaderElectionListenerInterfaceAdapter>> entry: _leaderChangeListeners.entrySet()) {
+          LOG.info("Resubscribe leader change listener for leaderPath {}.", entry.getKey());
+          String leaderPath = entry.getKey();
+          Set<LeaderElectionListenerInterfaceAdapter> listeners = entry.getValue();
+          for (LeaderElectionListenerInterfaceAdapter listener : listeners) {
+            _metaClient.subscribeDataChange(leaderPath + LEADER_ENTRY_KEY,
+                    listener, false /*skipWatchingNonExistNode*/); // we need to subscribe event when path is not there
+            _metaClient.subscribeStateChanges(listener);
+          }
+        }
         // touch leader node to renew session ID
```

Comment:
why not have a function for this and we can apply to:
1. First time start
2. reconnect.

I think they are on different code path.